wpt_commit_stats.py

```diff
@@ -0,0 +1,277 @@
+    WPT_ONLY = 'WPT only'
+    LEGACY_ONLY = 'legacy only'
+    MIXED = 'mixed'
+
+    def __init__(self):
+        self.commits = []
+        self.wpt_only_commits = []
+        self.legacy_only_commits = []
+
+    @staticmethod
+    def check_revision(git, rev):
+        files = git.run('diff-tree', '--diff-filter='+DIFF_FILTER,
+                        '--no-commit-id', '--name-only', '-r',
+                        rev, WEB_TESTS_DIR)
+        uses_wpt = False
+        uses_legacy = False
+        for f in files.splitlines():
+            if any([r.match(f) for r in _NOT_TEST_EXPRS]):
+                continue
+            if f.startswith(WPT_DIR):
+                uses_wpt = True
+            else:
+                uses_legacy = True
+        if not uses_wpt and not uses_legacy:
+            return CommitStats.NO_TESTS
+        if uses_wpt and not uses_legacy:
+            return CommitStats.WPT_ONLY
+        if not uses_wpt and uses_legacy:
+            return CommitStats.LEGACY_ONLY
+        return CommitStats.MIXED
+
+    def accumulate_result(self, rev, result):
+        if result == self.NO_TESTS:
+            return
+        self.commits.append(rev)
+        if result == self.WPT_ONLY:
+            self.wpt_only_commits.append(rev)
+        elif result == self.LEGACY_ONLY:
+            self.legacy_only_commits.append(rev)
+        else:
+            assert result == self.MIXED, 'Unknown result: ' + str(result)
+
+    def get_result(self):
+        return self.commits, self.wpt_only_commits, self.legacy_only_commits
+
+    def get_stats(self):
+        if len(self.commits) == 0:
+            # Avoid division by zero.
+            return 'no commits with web tests found'
+        return 'total: %d, WPT-only: %d (%.2f), legacy-only: %d (%.2f)' % (
+            len(self.commits),
+            len(self.wpt_only_commits),
+            float(len(self.wpt_only_commits)) / len(self.commits),
+            len(self.legacy_only_commits),
+            float(len(self.legacy_only_commits)) / len(self.commits),
+        )
+
+
+class FileStats(AbstractCheck):
+    @staticmethod
+    def _make_dict():
+        return {i: set() for i in DIFF_FILTER}
+
+    def __init__(self):
+        self.wpt = self._make_dict()
+        self.legacy = self._make_dict()
+
+    @staticmethod
+    def check_revision(git, rev):
+        output = git.run('diff-tree',
+                         '--no-commit-id', '--name-status', '-r',
+                         '--diff-filter='+DIFF_FILTER,
+                         '-C30', '-M', '--find-copies-harder',
+                         rev, WEB_TESTS_DIR)
+        wpt = FileStats._make_dict()
+        legacy = FileStats._make_dict()
+        for line in output.splitlines():
+            # status, filename OR status, original filename, new filename
+            parts = line.split(maxsplit=2)
+            # status could be "Cnn" where nn is the ratio.
+            status = parts[0][0]
+            f = parts[-1]
```

Comment:
Can you comment that `parts[1]` won't be used for anything in the "status, original filename, new filename" case and why that's OK?